src/init.c

```diff
@@ -71,21 +71,17 @@ void jl_init_stack_limits(int ismaster, void **stack_lo, void **stack_hi)
         size_t stacksize;
         pthread_attr_getstack(&attr, &stackaddr, &stacksize);
         pthread_attr_destroy(&attr);
-        *stack_lo = (void*)stackaddr;
-#pragma GCC diagnostic push
-#if defined(_COMPILER_GCC_) && __GNUC__ >= 12
-#pragma GCC diagnostic ignored "-Wdangling-pointer"
-#endif
-        *stack_hi = (void*)__builtin_frame_address(0);
-#pragma GCC diagnostic pop
+        *stack_hi = stackaddr;
+        *stack_lo = (char*)stackaddr - stacksize;
         return;
 #  elif defined(_OS_DARWIN_)
         extern void *pthread_get_stackaddr_np(pthread_t thread);
         extern size_t pthread_get_stacksize_np(pthread_t thread);
         pthread_t thread = pthread_self();
         void *stackaddr = pthread_get_stackaddr_np(thread);
-        *stack_lo = (void*)stackaddr;
-        *stack_hi = (void*)__builtin_frame_address(0);
+        size_t stacksize = pthread_get_stacksize_np(thread);
+        *stack_hi = stackaddr;
```

Comment:
Don't you meant to switch these, since the malloc region is stackaddr to stackaddr+stacksize?

Reply:
Maybe I am misunderstanding you (sorry), but: stacks on all OS/CPU combos supported by Julia grow downwards, the stack starts at the highest address `stackaddr` and grows downwards towards `stackaddr-stacksize` (right after which the guard pages comes by default, at least for pthread created stacks with default settings).

I double checked this in darwin_libpthread, glibc and muslc and via real-world experiments on darwin/macos and linux+glibc (no muslc for me

Reply:
@vtjnash does my explanation make sense to you? Any other questions?

Reply:
Okay, yeah, it is confusing that it is inverted from the values returned by `pthread_attr_getstack`, but apparently that is the legacy behavior